### PR TITLE
(maint) run golangci-lint in ci

### PR DESCRIPTION
**Pull request by tphoney**
part of https://github.com/overmindtech/workspace/issues/105

### CODE REVIEW

**Comment by tphoney** [comment]
i separated out the possibly controversial changes into a separate commit.

File: go.mod [hunk 1]
```diff
@@ -1,6 +1,6 @@
 module github.com/overmindtech/aws-source
 
-go 1.22.5
+go 1.22.4
```
Comment (dylanratcliffe) [inline go.mod:3]: Why are downgrading go?

**Review by dylanratcliffe — APPROVED** [review]
Just the one query

File: go.mod (hunk 1 above)
Comment (tphoney) [reply go.mod:3]: one of lifes great mysteries.  :)

go.work in workspace was really angry about being 1.22.4 and this go.mod it 1.22.5

Comment (DavidS-ovm) [reply go.mod:3]: I updated the workspace yesterday to use 1.22.5 everywhere because riverqueue pulled us up. please just fetch/rebase everything

Comment (tphoney) [reply go.mod]: fixed, not sure why i was missing it. kicked the workspace folder a few times.

Comment (DavidS-ovm) [reply go.mod]: one of lifes great mysteries.

**Comment by dylanratcliffe** [comment]
LGTM!

File: cmd/root.go [hunk 2]
```diff
@@ -258,7 +267,10 @@ func init() {
 	rootCmd.PersistentFlags().String("run-mode", "release", "Set the run mode for this service, 'release', 'debug' or 'test'. Defaults to 'release'.")
 
 	// Bind these to viper
-	viper.BindPFlags(rootCmd.PersistentFlags())
+	err = viper.BindPFlags(rootCmd.PersistentFlags())
+	if err != nil {
+		log.WithError(err).Fatal("could not bind flags to viper")
+	}
```
Comment (DavidS-ovm) [inline cmd/root.go:273]:
```suggestion
	cobra.CheckErr(viper.BindPFlags(rootCmd.PersistentFlags()))
```

cobra provides a short-cut for fatal viper initialisation errors. These always indicate programming errors and should be caught during testing.

File: sources/cloudfront/realtime_log_configs.go [hunk 3]
```diff
@@ -107,7 +107,7 @@ func NewRealtimeLogConfigsSource(client *cloudfront.Client, accountID string) *s
 			logConfigs := make([]*types.RealtimeLogConfig, 0, len(out.RealtimeLogConfigs.Items))
 
 			for _, logConfig := range out.RealtimeLogConfigs.Items {
-				logConfigs = append(logConfigs, &logConfig)
+				logConfigs = append(logConfigs, &logConfig) // nolint:exportloopref // this is resolved in go 1.22 https://github.com/kyoh86/exportloopref
```
Comment (DavidS-ovm) [inline sources/cloudfront/realtime_log_configs.go:110]: We're never gonna run a go version that is older than 1.22; let's add this ignore to the the default configs.

File: sources/directconnect/direct-connect-gateway.go [hunk 4]
```diff
@@ -26,7 +26,7 @@ func directConnectGatewayOutputMapper(_ context.Context, cli *directconnect.Clie
 
 	if len(resourceARNs) > 0 {
 		// get tags for the resources in a map by their ARNs
-		tags, err = arnToTags(context.Background(), cli, resourceARNs)
+		tags, err = arnToTags(ctx, cli, resourceARNs)
```
Comment (DavidS-ovm) [inline sources/directconnect/direct-connect-gateway.go:29]: :man_cook: :kiss:

**Review by DavidS-ovm — APPROVED** [review]
a few minor comments

**Pull request merged**